builder/builder.go

```diff
@@ -29,27 +31,100 @@ type Builder struct {
 	privKey    *ecdsa.PrivateKey
 
 	bc       *blockchain.Blockchain
+	pool     *mempool.Pool
 	vm       vm.VM
 	newHeads *feed.Feed[*core.Header]
 	log      utils.Logger
+
+	pendingLock  stdsync.Mutex
+	pendingBlock blockchain.Pending
+	headState    core.StateReader
+	headCloser   blockchain.StateCloser
 }
 
-func New(privKey *ecdsa.PrivateKey, ownAddr *felt.Felt, bc *blockchain.Blockchain, builderVM vm.VM, log utils.Logger) *Builder {
+func New(privKey *ecdsa.PrivateKey, ownAddr *felt.Felt, bc *blockchain.Blockchain, pool *mempool.Pool, builderVM vm.VM, log utils.Logger) *Builder {
 	return &Builder{
 		ownAddress: *ownAddr,
 		privKey:    privKey,
 
 		bc:       bc,
+		pool:     pool,
 		vm:       builderVM,
 		newHeads: feed.New[*core.Header](),
+		log:      log,
 	}
 }
 
 func (b *Builder) Run(ctx context.Context) error {
```

Comment:
Maybe add a comment / documentation for this? Eg "Mempool transactions are continuously being popped, and run against the pending state. Every N seconds the pending state is applied to the blockchain, and then reset."

Reply:
It will most probably evolve a lot really fast. I am leaving documentation for the later stages.